cmake/modules/FindIconv.cmake

```diff
@@ -0,0 +1,40 @@
+# FindIconv
+# ----------
+# Finds the Iconv library
+#
+# This will define the following variables:
+#
+#  ICONV_FOUND - system has Iconv
+#  ICONV_INCLUDE_DIR - the Iconv include directory
+#  ICONV_LIBRARIES - Link these to use Iconv
+
+check_function_exists(iconv_open ICONV)
+
+if(ICONV)
+  # iconv functions were found in libc
+  find_package_handle_standard_args(Iconv
+                                  REQUIRED_VARS ICONV)
+
+else()
+  # check for libiconv
+  find_path(ICONV_INCLUDE_DIR iconv.h)
+
+  find_library(ICONV_LIBRARY NAMES iconv libiconv libiconv-2)
+
+  if(ICONV_LIBRARY)
+    set(CMAKE_REQUIRED_LIBRARIES ${ICONV_LIBRARY})
+    check_function_exists(iconv_open ICONV)
+  endif()
+
+  if(ICONV)
+    set(ICONV_INCLUDE_DIRS ${ICONV_INCLUDE_DIR})
+    set(ICONV_LIBRARIES ${ICONV_LIBRARY})
+  endif()
+
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(Iconv
+                                  REQUIRED_VARS ICONV_LIBRARY ICONV_INCLUDE_DIR)
```

Comment:
and add ICONV here. It might also make sense to rename Iconv to something not possibly clashing with the upper case var ICONV. In that case adjust the ICONV_FOUND block mentioned in my comment above.